src/lib/testing/cypress/mock/README.md

```diff
@@ -0,0 +1,298 @@
+  that will record the response XHR (sanitized) to an internal data structure, mapped by URL/method. If mocking is to be performed,
+  it then loads fixtures in the order specified above. If mocking data, and hence mocking login, this method visits '/' to kick off
+  the test. Otherwise, it performs the single sign-on login.
+
+- tearDown(): If recording, this method dumps the URL/method to response data map to a _mockdata.json_ file in the
+  _cypress/recordings/testname/date_ directory. It then logs the user out.
+
+#### Usage
+
+Defining an E2E test
+
+```javascript
+import { MockUtility, TestType } from '@td-vantage/ui-platform/testing/cypress';
+
+describe('E2E Test', () => {
+  const mockUtil: MockUtility = new MockUtility();
+
+  before(() => {
+    mockUtil.setUp('mye2etest', TestType.e2e, ['golden'], 'user1');
+  });
+```
+
+Defining a unit test
+
+```javascript
+import { MockUtility, TestType } from '@td-vantage/ui-platform/testing/cypress';
+
+describe('Unit Test', () => {
+  const mockUtil: MockUtility = new MockUtility();
+
+  before(() => {
+    mockUtil.setUp('myunittest', TestType.unit, ['empty'], 'user1');
+  });
+```
+
+### Unit Testing
+
+Unit test specs should reside in the `cypress/unit` directory.
+
+Unit tests are written to rely solely on mock data. As mentioned above, data can be mocked by specifying fixtures that will be
+automatically loaded by the infrastructure or by defining routes in-line in the spec.
+
+There are template fixtures in the cypress/fixtures/templates directory to help you start mocking calls that are invoked on display
+of the homepage. Or you may choose to define your own test specific fixture in your test specific directory or construct routes
+inline in your test code.
+
+### E2E Testing
+
+E2E test specs should reside in the `cypress/e2e` directory.
+
+E2e tests should be written to exercise user flows and examine DOM elements based on the content of response data. This allows
+e2e tests to be agnostic about the environment within which they are run. You can, and probably should, define a template to be
+utilized for use in the checkin pipeline.
+
+### Recording Fixtures
+
+The testing framework has a switch to allow recording XHR response data into _mockdata.json_ fixture files. Enable recording by:
+
+- Run the tests by using the `npm run cypress:runrecord` script.
+- Run Cypress using the `npm run cypress:openrecord` script and then run individual tests.
+- Modify the call to the `setUp` method in a specific test to pass the optional fourth "record" parameter as `true`.
+
+Recorded response fixtures are written to the _cypress/recordings/testname/date_ directory.
+
+## Running Tests
+
+The following scripts exist in the package.json to facilitate running Cypress tests. All require that you
+start the web server in a separate command window.
+
+cypress:open - Open the Cypress dashboard
```

Comment:
Let's make sure this is properly formatted markdown

Reply:
Done